### PR TITLE
Upgrade indirect dependancy go-libp2p-core to v0.20.1

### PR DESCRIPTION
**Pull request by bsalunke**
**Upgrade indirect dependancy go-libp2p-core v0.8.6 to go-libp2p-core v0.20.1 to remove indirect dependacy of btcd module which is vulnerable to CVE-2022-44797**

**What is it fixing**
Nerdctl has an indirect dependency module ("github.com/btcsuite/btcd v0.21.0-beta) having vulnerability reported  https://nvd.nist.gov/vuln/detail/CVE-2022-44797. 

As explained by @AkihiroSuda, it's false positive and nerdctl does not trigger the code path in any way to invoke "btcd" module. The fix provided was VEX set to eliminate https://github.com/containerd/nerdctl/issues/1571

However APIK some of the open source scanner does not support VEX as of today. e.g. https://github.com/aquasecurity/trivy/issues/1836. 
 

**Fix**
Upgrade indirect dependancy go-libp2p-core v0.8.6 to go-libp2p-core v0.20.1

Please let me know if more information needs to be added.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -58,11 +58,16 @@ require (
...
+	github.com/mattn/go-pointer v0.0.1 // indirect
+)
+
 require (
 	github.com/AdaLogics/go-fuzz-headers v0.0.0-20221007124625-37f5449ff7df // indirect
 	github.com/AdamKorcz/go-118-fuzz-build v0.0.0-20220912195655-e1f97a00006b // indirect
 	github.com/Microsoft/hcsshim v0.10.0-rc.1 // indirect
-	github.com/btcsuite/btcd v0.21.0-beta // indirect
 	// ↑The `github.com/btcsuite/btcd` line exists for the indirect dependency on `github.com/btcsuite/btcd/btcec` (secp256k1 elliptic curve cryptography library) via `github.com/ipfs/go-ipfs-http-client`.
 	// https://github.com/btcsuite/btcd/tree/master/btcec
```
Comment (AkihiroSuda) [inline go.mod:72]: This note can be now dropped

File: go.mod [hunk 2]
```diff
@@ -58,11 +58,16 @@ require (
 	gotest.tools/v3 v3.4.0
 )
 
+require (
```
Comment (AkihiroSuda) [inline go.mod:61]: No need to create another require()

File: go.mod (hunk 1 above)
Comment (bsalunke) [reply go.mod]: Updated PR to address review comment.

**Comment by AkihiroSuda** [comment]
Could you squash the commits? Then LGTM

**Comment by AkihiroSuda** [comment]
Thanks, but please sign the commit for DCO
https://github.com/apps/dco

(run `git commit -a -s --amend`, and make sure that the `Signed-off-by: NAME <EMAIL>` line with your real name is included in the commit message)

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by bsalunke** [comment]
@AkihiroSuda  Thank you for help. I see it's marked for v1.1.0. Any rough timeline for v1.1.0 release?

**Comment by AkihiroSuda** [comment]
> @AkihiroSuda Thank you for help. I see it's marked for v1.1.0. Any rough timeline for v1.1.0 release?

As soon as https://github.com/containerd/accelerated-container-image/issues/155#issuecomment-1343866742 is addressed

**Pull request merged**